src/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/Hosting/TypeCatalogTests.cs

```diff
@@ -0,0 +1,574 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition.Hosting;
+using System.ComponentModel.Composition.Factories;
+using System.ComponentModel.Composition.Primitives;
+using System.ComponentModel.Composition.UnitTesting;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.UnitTesting;
+using Microsoft.Internal;
+
+using System.Globalization;
+using System.Text;
+using Xunit;
+
+namespace System.ComponentModel.Composition
+{
+    [Export]
+    public class  TypeCatalogTestsExporter {}
+
+// This is a glorious do nothing ReflectionContext
+    public class TypeCatalogTestsReflectionContext : ReflectionContext
+    {
+        public override Assembly MapAssembly(Assembly assembly)
+        {
+            return assembly;
+        }
+
+#if FEATURE_INTERNAL_REFLECTIONCONTEXT
```

Comment:
We don't define any defines, generally. So for each one you must remove it, and make a decision whether to keep the code inside it. If they existed in the implementation, do the same thing we did for that. There shold probably be zero `#if` in the code.